packages/eslint-plugin/lib/utils/patterns.ts

```diff
@@ -1,40 +1,45 @@
-export const sourceFiles = [
-    "*.[jt]s?(x)",
-    "*.[cm]js"
+export const javaScriptFiles = [
+    "**/*.js?(x)",
+    "**/*.[cm]js"
+];
+
+export const typeScriptFiles = [
+    "**/*.ts?(x)"
 ];
 
-export const typescriptFiles = [
-    "*.ts?(x)"
+export const sourceFiles = [
+    ...javaScriptFiles,
+    ...typeScriptFiles
 ];
 
 export const testFiles = [
-    "*.test.[jt]s",
-    "*-test.[jt]s",
+    "**/*.test.[jt]s",
```

Comment:
You mentioned that you were going to revisit this, did you?

Reply:
I did revisit this, they need the ** or they will only lint the top level files.